server/handlers/component_handler.go

```diff
@@ -1204,3 +1204,97 @@ func (h *Handler) RegisterMeshmodelComponents(rw http.ResponseWriter, r *http.Re
 	}
 	go h.config.MeshModelSummaryChannel.Publish()
 }
+
+// swagger:route GET /api/meshmodels/registrants GetMeshmodelRegistrants
+// Handle GET request for getting all meshmodel registrants
+//
+// # Returns a list of registrants and their respective models components and relationships
+//
+// ```?page={page-number}``` Default page number is 1
+//
+// ```?search={Hostaname}``` Gets host by the name
+//
+// ```?pagesize={pagesize}``` Default pagesize is 25. To return all results: ```pagesize=all```
+// responses:
+//
+//	200: []MesheryHostsContextPage
+
+func (h *Handler) GetMeshmodelRegistrants(rw http.ResponseWriter, r *http.Request) {
+	rw.Header().Add("Content-Type", "application/json")
+	enc := json.NewEncoder(rw)
+
+	limitstr := r.URL.Query().Get("pagesize")
+	pagestr := r.URL.Query().Get("page")
+	searchQuery := r.URL.Query().Get("search")
+
+	limit, _ := strconv.Atoi(limitstr)
+	if limitstr != "all" && limit == 0 {
+		limit = DefaultPageSizeForMeshModelComponents
+	}
+
+	page, _ := strconv.Atoi(pagestr)
+	if page <= 0 {
+		page = 1
+	}
+
+	offset := (page - 1) * limit
+
+	hosts, hostsCount, err := h.registryManager.GetHostsForModels(h.dbHandler.DB, limit, offset, searchQuery)
+	if err != nil {
+		h.log.Error(ErrGetMeshModels(err))
+		http.Error(rw, ErrGetMeshModels(err).Error(), http.StatusInternalServerError)
+	}
+
+	result := make([]MesheryHostsDisplay, 0, len(hosts))
+
+	for _, host := range hosts {
```

Comment:
Why do we need to loop through each `host` entry?
We can write a single query to join hosts and registries table. (the reason I'm asking to limit the use of go and rather use SQL is because very soon the performance of these handlers degrades when the number of hosts would increase. We have seen this in the past).

Reply:
Here is the query that you can use:

```
SELECT
    h.id AS host_id,
    h.hostname,
    r.type AS registry_type,
    COUNT(*) AS entity_count
FROM
    hosts h
LEFT JOIN
    registries r ON h.id = r.registrant_id
GROUP BY
    h.id, h.hostname, r.type
ORDER BY
    h.id, r.type;

```

You can create a single handler  to get summary of hosts. Pass the necessary info to the handler using the filter option (I mentioned [here](https://github.com/meshery/meshkit/pull/383/files#r1354956317)).

Reply:
Also, pre-defining all the types of resources in golang would increase the number of places that requires changes when a new type is defined. Ideally we should use whichever types are available in the DB